modules/account/src/main/kotlin/edu/stanford/spezi/module/account/register/RegisterFormValidator.kt

```diff
@@ -36,27 +33,24 @@ class RegisterFormValidator @Inject internal constructor() {
     fun isFormValid(uiState: RegisterUiState): Boolean {
         val passwordConditionSatisfied = {
             if (uiState.isGoogleSignUp) {
-                passwordResult(uiState.password.value) is Result.Valid
+                isValidPassword(uiState.password.value).isValid
             } else {
                 true
             }
         }
-        return emailResult(uiState.email.value) is Result.Valid &&
-            firstnameResult(uiState.firstName.value) is Result.Valid &&
-            lastnameResult(uiState.lastName.value) is Result.Valid &&
-            isGenderValid(uiState.selectedGender.value) is Result.Valid &&
-            birthdayResult(uiState.dateOfBirth) is Result.Valid &&
+        return isValidEmail(uiState.email.value).isValid &&
+            firstnameResult(uiState.firstName.value).isValid &&
+            lastnameResult(uiState.lastName.value).isValid &&
+            isGenderValid(uiState.selectedGender.value).isValid &&
+            birthdayResult(uiState.dateOfBirth).isValid &&
             passwordConditionSatisfied()
     }
 
-    sealed interface Result {
-        data object Valid : Result
-        data class Invalid(val message: String) : Result
-
-        fun errorMessageOrNull() = if (this is Invalid) message else null
-    }
-
-    private companion object {
-        const val MIN_PASSWORD_LENGTH = 6 // Minimum for firebase
+    fun isRegisterButtonEnabled(uiState: RegisterUiState): Boolean {
```

Comment:
This method should belong to the view model, validator is responsible to validate the form and not to decide whether the button should be enabled or not ;)